### PR TITLE
Paul submission - main criteria

### PR DESCRIPTION
**Pull request by therearedoors**

### CODE REVIEW

File: src/client/App.jsx [hunk 1]
```diff
@@ -14,16 +20,54 @@ function App() {
...
+  const handleRegister = async user => {
+    fetch(`${apiUrl}/user/register`, {
+      ...headers,
+      body: JSON.stringify(user)
+    })
+    .then(res => res.json())
+    .then(json => {
+      console.log(json)
+      if (json.error) {
+        window.alert(`${json.error}!`)
```
Comment (JasonKundi) [inline src/client/App.jsx:32]: Like the inclusion of alerts. Although it's a small implementation it shows you're thinking about display in a real life application as opposed to just meeting the requirements

File: src/client/App.jsx [hunk 2]
```diff
@@ -14,16 +20,54 @@ function App() {
...
+      if (json.data) {
+        localStorage.setItem('token',json.data)
+        window.alert(`${json.data} token!`)
+      }
+    })
   };
   
-  const handleCreateMovie = async ({ title, description, runtimeMins }) => {
-    
+  const handleCreateMovie = async (movie) => {
```
Comment (JasonKundi) [inline src/client/App.jsx:60]: I like how you've constructed this handleCreateMovie function. The way you've simplified the fetch by creating the headersWithAuth variable is very neat.

File: src/client/components/MovieForm.jsx [hunk 3]
```diff
@@ -6,15 +6,16 @@ export default function MovieForm({ handleSubmit }) {
...
+        setMovie(currentMovie => ({ title: '', description: '', runtimeMins: 60 }));
     }
 
     const handleChange = (e) => {
         const { name, value } = e.target;
 
-        setMovie({
+        setMovie(currentMovie => ({
             ...movie,
             [name]: name === 'runtimeMins' ? parseInt(value) : value
```
Comment (Willcowen) [inline src/client/components/MovieForm.jsx:17]: Nice use of short hand syntax!

File: src/server/controllers/user.js [hunk 4]
```diff
@@ -1,35 +1,46 @@
...
+        const createdUser = await prisma.user.create({
+        data: {
+            username: username,
+            password: hash
+        }
+        })
+    console.log("done")
     res.json({ data: createdUser });
+    } catch (e) {
+        if (e instanceof Prisma.PrismaClientKnownRequestError){
```
Comment (Willcowen) [inline src/server/controllers/user.js:21]: Interesting, so this evaluates true of the error is an instance of the prisma object?

Comment (therearedoors) [reply src/server/controllers/user.js:21]: As I understand it, Prisma has it's own type of error object - PrismaClientKnownRequestError: https://www.prisma.io/docs/concepts/components/prisma-client/handling-exceptions-and-errors

So this code catches any errors that Prisma throws up - this stops e.g. the server crashing when the client tries to add a movie that's already in the database.

**Review by alsyoufomar — COMMENTED** [review]
I like the idea of using new techniques every time, well done Paul!

File: src/client/App.jsx (hunk 2 above)
Comment (therearedoors) [reply src/client/App.jsx:60]: yeah I just didn't want to copy paste more fetch request boilerplate, this seems like the sort of thing someone could write a library/middleware for to tidy it up :)

File: src/client/App.jsx [hunk 5]
```diff
@@ -14,16 +20,54 @@ function App() {
...
+    let headersWithAuth = headers
+    headersWithAuth.headers.authorization = await localStorage.getItem('token')
+    console.log(headersWithAuth)
+    await fetch(`${apiUrl}/movie`, {
+      ...headersWithAuth,
+      body: JSON.stringify(movie)
+    })
+    await fetch(`${apiUrl}/movie`)
```
[the remark this answers is not in the document]
Comment (therearedoors) [reply src/client/App.jsx:68]: When I read the 'update movie list for the client', my first thought was just to make another getMovies request, but it's possible and probably better here to just use the response from the successful post request and just update the client that way.

**Pull request closed without merging**